gammapy/utils/fitting/datasets.py

```diff
@@ -0,0 +1,50 @@
+import numpy as np
+from astropy.utils import lazyproperty
+from .parameter import Parameters
+
+
+class Datasets:
+    """Join multiple datasets
+
+    Parameters
+    ----------
+    datasets : `Dataset` or list of `Dataset`
+        List of `Dataset` objects ot be joined.
+    mask : `~numpy.ndarray`
+        Global fitting mask used for all datasets.
+
+    """
+    def __init__(self, datasets, mask=None):
+        if not isinstance(datasets, list):
+            datasets = [datasets]
+        self.datasets = datasets
+
+        if mask is not None and not self.is_all_same_type:
+            raise ValueError("Cannot apply mask if datasets are not of the same type.")
+
+        self.mask = mask
+
+        # join parameter lists
+        parameters = []
+        for dataset in self.datasets:
+            parameters += dataset.parameters.parameters
+        self.parameters = Parameters(parameters)
+
+    @property
+    def model(self):
+        return self.datasets[0].model
```

Comment:
I am a bit confused. The same `model` need not reside on all `datasets`, right? So what is the utility of this function?

Reply:
I decided to remove this again and break the backwards compatibility. The result object returned by `Fit.optimize()` does not contain a reference to the model anymore but instead to the joined parameter list. I think this is a justified change, because there is not necessarily
a single model across many datasets (e.g. if there are background model per dataset).